mobius/src/cline_arc.rs

```diff
@@ -54,117 +78,137 @@ pub struct ClineArc {
-                // in which case we just want a ray pair through a and c.
-                //
-                // we can just check a -> b and b -> c. If the dot product is
-                // positive, then the points are in order.
-                let in_order = Complex::dot(self.b - self.a, self.c - self.b) > 0.0;
-
-                if in_order {
-                    ClineArcGeometry::LineSegment(LineSegment {
-                        start: self.a,
-                        end: self.c,
-                    })
-                } else {
-                    let ac = (self.c - self.a).normalize().unwrap();
-                    ClineArcGeometry::ThruInfinity(DoubleRay(
-                        Ray {
-                            start: self.a,
-                            unit_dir: -ac,
-                        },
-                        Ray {
-                            start: self.c,
-                            unit_dir: ac,
-                        },
-                    ))
-                }
-            }
-            GeneralizedCircle::Circle(Circle { center, radius }) => {
-                // Determine if the 3 points circulate counterclockwise or
-                // clockwise by forming a triangle ABC and computing
-                // (the magnitude of) the wedge product.
-                let ab = self.b - self.a;
-                let ac = self.c - self.a;
-                let ccw = Complex::wedge(ab, ac) > 0.0;
-
-                let theta_a = (self.a - center).arg().unwrap();
-                let theta_c = (self.c - center).arg().unwrap();
-                let a_bigger = theta_a > theta_c;
-
-                let end_angle = if !ccw && !a_bigger {
-                    theta_c - TAU
-                } else if ccw && a_bigger {
-                    theta_c + TAU
-                } else {
-                    theta_c
-                };
-
-                let theta_b = (self.b - center).arg().unwrap();
-
-                ClineArcGeometry::CircularArc(CircularArc {
-                    circle: Circle { center, radius },
-                    angle_a: theta_a,
-                    angle_b: theta_b,
-                    angle_c: end_angle,
-                })
-            }
+            GeneralizedCircle::Line(_) => Ok(self.compute_line_geometry()),
+            GeneralizedCircle::Circle(circle) => self.compute_circle_geometry(circle),
         }
     }
 }
 
 impl From<CircularArc> for ClineArc {
     fn from(value: CircularArc) -> Self {
-        let CircularArc {
-            circle,
-            angle_a,
-            angle_b,
-            angle_c,
-        } = value;
-
+        let CircularArc { circle, angles } = value;
         let Circle { center, radius } = circle;
+        let ArcAngles(angle_a, angle_b, angle_c) = angles;
         let a = center + Complex::from_polar(radius, angle_a);
         let b = center + Complex::from_polar(radius, angle_b);
         let c = center + Complex::from_polar(radius, angle_c);
```

Comment:
Even if we remove the b parameter, computing the midpoint would allow this to still work